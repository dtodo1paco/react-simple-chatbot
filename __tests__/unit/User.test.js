jest.autoMockOff();

import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import { getRenderedContent, getComposite} from '../util/render.js';


const User = require('components/User').default;

describe('User testing...', () => {
    /** CUSTOM FOR COMPONENT **/
    const exampleUser = {
        username: "testun",
        password: "supersecret"
    };

    const handler = jest.fn();

    beforeEach(function() {

    });

    describe('render', () => {
        it('snapshot', () => {
            const wrapper = shallow(<User user={exampleUser} loading={false}
                loginHandler={handler}
                signupHandler={handler}
                logoutHandler={handler}
            />);
            expect(wrapper).toMatchSnapshot();
        });
        it('when loading is true, loading icon is shown', () => {
            let content = getRenderedContent(<User user={null} loading={true}
                loginHandler={handler}
                signupHandler={handler}
                logoutHandler={handler}
            />)
            expect(content.props.className).toBe('loading-icon');
        });
        it('when loading is false and no user given, a tab menu is shown', () => {
            let content = getRenderedContent(<User user={null} loading={false}
                loginHandler={handler}
                signupHandler={handler}
                logoutHandler={handler}
            />);
            expect(content.key).toBe('tabs-menu');
        });
        it('when loading is false and a user given, a wellcome page is shown', () => {
            let content = getRenderedContent(<User user={exampleUser} loading={false}
                loginHandler={handler}
                signupHandler={handler}
                logoutHandler={handler}
            />);
            expect(content.key).toBe('wellcome-page');
        });
    });
});

/**
 console.log("Errors: " + renderedDOM().querySelectorAll('.visible').length);
**/