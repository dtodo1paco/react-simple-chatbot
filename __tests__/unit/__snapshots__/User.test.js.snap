// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User testing... render snapshot 1`] = `
<div
  className="user"
>
  <div
    key="wellcome-page"
  >
    Bienvenido al chatbot veterinario donde podrás plantear tus dudas acerca de tus mascotas.
    <br />
    <br />
    Entrarás en el chat con el nombre: 
    <br />
    <br />
    <button
      className="iconButton"
      onClick={[MockFunction]}
      type="submit"
    >
      <span
        className="label"
      >
        Abandonar
      </span>
    </button>
  </div>
</div>
`;
